### ISSUE TITLE
Implement LSTM Betting Strategy

### ISSUE THREAD

**Issue by Acusick1** [Acusick1/odds#30]
**Goal**: Create a minimal LSTM betting strategy implementation that provides a clean, extensible framework for experimenting with time-series models. The focus is on correct integration with the backtesting system and ease of modification, not on finding optimal hyperparameters or achieving high performance.

**Complexity**: 🔴 Hard

**Location**:
- `packages/odds-analytics/odds_analytics/lstm_strategy.py` (new module)
- `tests/unit/test_lstm_strategy.py` (new test file)
- `pyproject.toml` (add PyTorch dependencies)

**Prerequisites**:
- Understanding of LSTM architecture and time-series classification
- Familiarity with PyTorch (tensors, nn.Module, training loops)
- Knowledge of the `BettingStrategy` base class contract
- Issue #29 must be completed (sequence data loader)

**Implementation Approach**:

Follow the `XGBoostStrategy` pattern exactly - the goal is a working reference implementation that's easy to modify:

```python
class LSTMStrategy(BettingStrategy):
    """
    LSTM-based betting strategy for time-series line movement patterns.
    
    This is a reference implementation optimized for experimentation and
    modification, not for performance. Use simple, readable architecture
    with sensible defaults that can be easily customized.
    """
    
    def __init__(
        self,
        model_path: str | None = None,
        lookback_hours: int = 72,
        timesteps: int = 24,
        hidden_size: int = 64,  # Simple default, easy to change
        num_layers: int = 2,
        dropout: float = 0.2,
        market: str = "h2h",
        min_edge_threshold: float = 0.03,
        min_confidence: float = 0.52,
        **kwargs
    ):
        """Initialize with reasonable defaults - user can experiment by changing."""
        
    def build_model(self) -> nn.Module:
        """
        Build simple LSTM architecture.
        
        Keep architecture simple and readable - user will modify this
        for experimentation, not optimize it here.
        """
        
    def train(
        self,
        X_train: np.ndarray,
        y_train: np.ndarray,
        masks_train: np.ndarray,
        X_val: np.ndarray | None = None,
        y_val: np.ndarray | None = None,
        masks_val: np.ndarray | None = None,
        epochs: int = 20,
        batch_size: int = 32,
        learning_rate: float = 0.001,
    ) -> dict[str, list[float]]:
        """
        Train LSTM with basic training loop.
        
        Returns training history for user to analyze/visualize.
        Keep implementation simple - user will add features like
        early stopping, learning rate scheduling, etc. as needed.
        """
        
    async def evaluate_opportunity(
        self,
        event: BacktestEvent,
        odds_snapshot: list[Odds],
        config: BacktestConfig,
    ) -> list[BetOpportunity]:
        """Predict using trained LSTM - main integration point."""
        
    def save_model(self, filepath: str) -> None:
        """Save model weights and config."""
        
    def load_model(self, filepath: str) -> None:
        """Load model weights and config."""
```

**Constraints**:
- Must inherit from `BettingStrategy` base class
- Must use PyTorch (not TensorFlow)
- Must integrate with `SequenceFeatureExtractor` via dependency injection
- Must implement async `evaluate_opportunity()` method
- Must save/load model weights for persistence
- Must use `BetOpportunity.confidence` field for Kelly Criterion bet sizing
- Must query historical odds using sequence data loader from Issue #29
- **Focus on code clarity and ease of modification over performance**
- Use simple, vanilla implementations (no complex optimizations)

**Scope**:
- Add PyTorch dependencies to `pyproject.toml` (`torch>=2.0.0`)
- Implement `LSTMStrategy` class following XGBoost pattern
- Build simple LSTM architecture (2 layers, 64 hidden units - easy to modify)
- Implement basic training loop (no early stopping or advanced features - user will add if needed)
- Implement `evaluate_opportunity()` with async odds querying
- Implement model persistence (save/load using torch.save)
- Handle variable-length sequences using attention masks from SequenceFeatureExtractor
- Write unit tests covering: build, train, save/load, predict workflow
- Add to `AVAILABLE_STRATEGIES` registry
- **DO NOT spend time tuning hyperparameters or optimizing architecture**
- **DO NOT implement advanced training features (early stopping, LR scheduling, etc.) - keep simple**

**Success Criteria**:
- [ ] `LSTMStrategy` correctly implements `BettingStrategy` interface
- [ ] LSTM model builds and trains successfully with default parameters
- [ ] Training method returns history dict with loss values for plotting
- [ ] `evaluate_opportunity()` queries historical odds and makes predictions
- [ ] Model save/load preserves weights and reproduces predictions
- [ ] Returns `BetOpportunity` with confidence scores
- [ ] Unit tests pass: build, train, save/load, predict, integration with backtesting
- [ ] Can be imported and used in backtesting engine
- [ ] Code is clean, well-commented, and easy to modify
- [ ] Added to `AVAILABLE_STRATEGIES` registry
- [ ] No ruff or type errors

**Related Files**:
- `packages/odds-analytics/odds_analytics/ml_strategy_example.py` - **Primary template**: Follow this structure exactly
- `packages/odds-analytics/odds_analytics/feature_extraction.py` - `SequenceFeatureExtractor` (lines 499-875)
- `packages/odds-analytics/odds_analytics/backtesting/services.py` - `BettingStrategy` base class
- `packages/odds-analytics/odds_analytics/strategies.py` - `AVAILABLE_STRATEGIES` registry
- `tests/unit/test_ml_strategy_example.py` - Test template

**Notes**:
- **This is infrastructure, not a production model** - prioritize clarity over performance
- Use simple vanilla PyTorch (no PyTorch Lightning, no complex features)
- Reasonable defaults are fine (2 layers, 64 hidden, lr=0.001) - user will experiment with these
- Don't spend time on hyperparameter tuning or architecture optimization
- The goal is a working LSTM that integrates cleanly with backtesting and is easy to modify
- User will handle experimentation and model improvement themselves
